### ISSUE TITLE
SSM Parameter shorthand incorrect syntax still not detected

### ISSUE THREAD

**Issue by gbudge** [aws-cloudformation/cfn-lint#4171]
### CloudFormation Lint Version

cfn-lint 1.36.0

### What operating system are you using?

MacOS

### Describe the bug

The issue raised in https://github.com/aws-cloudformation/cfn-lint/issues/4169 is still not resolved. Related PR: https://github.com/aws-cloudformation/cfn-lint/pull/4170

```shell
$ cfn-lint --version
cfn-lint 1.36.0
$ cfn-lint --format json --include-checks I -- "./4170-testing.yaml"
[]
$ cat ./4170-testing.yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: 'Example of missing incorrect shorthand syntax for SSM parameters.'

Resources:
  MyS3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: "{{ resolve:ssm:SomeParameter }}"
      Tags:
        - Key: CorrectSyntax
          Value: "{{resolve:ssm:SomeParameter}}" # This is correct shorthand syntax
        - Key: IncorrectSyntax
          Value: "{{ resolve:ssm:SomeParameter }}" # This is incorrect shorthand syntax
$ 
```

### Expected behavior

`BucketName: "{{ resolve:ssm:SomeParameter }}"` should be detected as invalid shorthand ssm parameter resolution.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: 'Example of missing incorrect shorthand syntax for SSM parameters.'

Resources:
  MyS3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: "{{ resolve:ssm:SomeParameter }}"
      Tags:
        - Key: CorrectSyntax
          Value: "{{resolve:ssm:SomeParameter}}" # This is correct shorthand syntax
        - Key: IncorrectSyntax
          Value: "{{ resolve:ssm:SomeParameter }}" # This is incorrect shorthand syntax
```

**Comment by kddejong** [comment]
You didn't wait for the next release.
